lib/src/model/puzzle/puzzle_service.dart

```diff
@@ -99,12 +104,29 @@ class PuzzleService {
   Future<PuzzleContext?> solve({
     required UserId? userId,
     required PuzzleSolution solution,
+    required Puzzle puzzle,
     PuzzleTheme angle = PuzzleTheme.mix,
   }) async {
     final data = await storage.fetch(
       userId: userId,
       angle: angle,
     );
+    final history = await historyStorage.fetch(
```

Comment:
The fetch and concatenation should be handled by the storage, in a transaction. Here we should only do:
```dart
await historyStorage.save(
  userId: userId,
  angle: angle,
  puzzleResult: puzzleResult
);
```